addons/crm/models/crm_lead.py

```diff
@@ -745,6 +771,18 @@ def create(self, vals_list):
                 vals['website'] = self.env['res.partner']._clean_website(vals['website'])
         leads = super().create(vals_list)
 
+        commercial_partner_ids = [vals['commercial_partner_id'] for vals in vals_list if vals.get('commercial_partner_id')]
+        CommercialPartners = self.env['res.partner'].with_prefetch(commercial_partner_ids)
+        for lead, lead_vals in zip(leads, vals_list):
+            if not lead_vals.get('partner_id') and lead_vals.get('commercial_partner_id'):
+                if 'default_partner_id' in self.env.context:
```

Comment:
We could call ``default_get(['partner_id'])['partner_id']`` to correctly support all default use cases